packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart

```diff
@@ -282,60 +284,99 @@ class ClassesTableDiff extends StatelessWidget {
     required this.classFilterButton,
     required this.before,
     required this.after,
+    required this.sizeTypeToShowForDiff,
   }) : super(key: key);
 
   final List<DiffClassStats> classes;
   final ValueNotifier<DiffClassStats?> selection;
   final AdaptedHeapData before;
   final AdaptedHeapData after;
+  final ValueNotifier<SizeType> sizeTypeToShowForDiff;
+
+  List<ColumnGroup> _columnGroups(SizeType sizeType, BuildContext context) {
+    final sizeTitle = maybeWrapWithTooltip(
+      child: Padding(
+        padding: const EdgeInsets.all(densePadding),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            RoundedDropDownButton<SizeType>(
+              isDense: true,
+              value: sizeType,
+              onChanged: (SizeType? value) =>
+                  sizeTypeToShowForDiff.value = value!,
+              items: SizeType.values
+                  .map(
+                    (sizeType) => DropdownMenuItem<SizeType>(
+                      value: sizeType,
+                      child: Text(sizeType.displayName),
+                    ),
+                  )
+                  .toList(),
+            ),
+            const SizedBox(width: denseSpacing),
+            const Text('Size'),
+          ],
+        ),
+      ),
+      tooltip: '${sizeType.displayName} size:\n${sizeType.description}',
+    );
 
-  static final _columnGroups = [
-    ColumnGroup.fromText(
-      title: '',
-      range: const Range(0, 1),
-    ),
-    ColumnGroup.fromText(
-      title: 'Instances',
-      range: const Range(1, 5),
-      tooltip: nonGcableInstancesColumnTooltip,
-    ),
-    ColumnGroup.fromText(
-      title: 'Shallow Dart Size',
-      range: const Range(5, 9),
-      tooltip: shallowSizeColumnTooltip,
-    ),
-    ColumnGroup.fromText(
-      title: 'Retained Dart Size',
-      range: const Range(9, 13),
-      tooltip: retainedSizeColumnTooltip,
-    ),
-  ];
+    return [
+      ColumnGroup.fromText(
+        title: '',
+        range: const Range(0, 1),
+      ),
+      ColumnGroup.fromText(
+        title: 'Instances',
+        range: const Range(1, 5),
+        tooltip: nonGcableInstancesColumnTooltip,
+      ),
+      if (sizeType == SizeType.shallow)
```

Comment:
Can we remove these conditions and just use a single `ColumnGroup` for `sizeTitle`?